### PR TITLE
Migrate S3329: Scaffold the rule for the new SE 

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**
Part of #7559

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InitializationVectorShouldBeRandom.CSharp8.cs [hunk 1]
```diff
@@ -0,0 +1,159 @@
+﻿using System;
```
Comment (mary-georgiou-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InitializationVectorShouldBeRandom.CSharp8.cs:1]: Here I practically added any test with a `using` statement.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/InitializationVectorShouldBeRandomTest.cs [hunk 2]
```diff
@@ -18,51 +18,102 @@
...
+        [Ignore] // ToDo: Remove after S3329 implementation
+        [TestMethod]
+        public void InitializationVectorShouldBeRandom_Roslyn_CSharp8() =>
+            roslynCS.AddPaths("InitializationVectorShouldBeRandom.CSharp8.cs")
+                .WithOptions(ParseOptionsHelper.FromCSharp8)
+                .Verify();
+
         [TestMethod]
-        public void InitializationVectorShouldBeRandom_DoesNotRaiseIssuesForTestProject() =>
+        public void InitializationVectorShouldBeRandom_Sonar_DoesNotRaiseIssuesForTestProject() =>
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/SymbolicExecution/InitializationVectorShouldBeRandomTest.cs:63]:
```suggestion
        public void InitializationVectorShouldBeRandom_DoesNotRaiseIssuesForTestProject_Sonar() =>
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InitializationVectorShouldBeRandom.CSharp8.cs [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Tests.Diagnostics
+{
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InitializationVectorShouldBeRandom.CSharp8.cs:6]: Kill the noise! :D
```suggestion
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InitializationVectorShouldBeRandom.cs [hunk 4]
```diff
@@ -0,0 +1,212 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Cryptography;
+
+namespace Tests.Diagnostics
+{
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InitializationVectorShouldBeRandom.cs:6]: Kill the noise!
```suggestion
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InitializationVectorShouldBeRandom.cs [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+            using (SymmetricAlgorithm sa = SymmetricAlgorithm.Create("AES"))
+            {
+                ICryptoTransform noParams = sa.CreateEncryptor(); // Compliant - IV is automatically generated
+                var defaultKeyAndIV = sa.CreateEncryptor(sa.Key, sa.IV); // Compliant
+
+                sa.GenerateKey();
+                var generateIVNotCalled = sa.CreateEncryptor(sa.Key, sa.IV);
+                var constantVector = sa.CreateEncryptor(sa.Key, initializationVectorConstant); // // FIXME Non-compliant  {{Use a dynamically-generated, random IV.}}
+//                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (Tim-Pohlmann) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InitializationVectorShouldBeRandom.cs:21]:
```suggestion
//                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by Tim-Pohlmann — CHANGES_REQUESTED** [review]
Some minor changes required.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InitializationVectorShouldBeRandom.CSharp8.cs (hunk 3 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/InitializationVectorShouldBeRandom.CSharp8.cs:6]: 🪓

**Review by Tim-Pohlmann — APPROVED** [review]
LGTM.
@pavel-mikula-sonarsource please merge this commit. The quality gate fails because this is a scaffolding PR.

**Pull request merged**